### PR TITLE
Change templating language from Chameleon to Jinja2

### PR DESCRIPTION
**Pull request by nathanbegbie**

### CODE REVIEW

**Comment by nathanbegbie** [comment]
@milton pls review

File: cms/templates/404.jinja2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{% extends "base.jinja2" %}
+{% import 'atoms.html' as atoms %}
+{% import 'molecules.html' as molecules %}
+{% import 'organisms.html' as organisms %}
+
+{% block content %}
+
+<h1>We can't find that page, sorry.</h1>
+Go to the<a href="/spice/">Home Page</a>
```
Comment (miltontony) [inline cms/templates/404.jinja2]: Please use `${request.route_url('home_jinja')}` for urls.

File: cms/templates/category.jinja2 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{% extends "base.jinja2" %}
+{% import 'atoms.html' as atoms %}
+{% import 'molecules.html' as molecules %}
+{% import 'organisms.html' as organisms %}
+
+{% block content %}
+
+{{molecules.image_and_heading(view, category)}}
+
+{{organisms.list_of_links_FFL(pages, category)}}
```
Comment (miltontony) [inline cms/templates/category.jinja2]: Not sure how you guys did stuff on _fufu_ but it doesn't feel right to have app-specific macros here. It feels like this should just be `{{organisms.list_of_links(pages, category)}}` or `{{organisms.list_of_links_menu(pages, category)}}`

File: cms/templates/content.jinja2 [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{% extends "base.jinja2" %}
+{% import 'atoms.html' as atoms %}
+{% import 'molecules.html' as molecules %}
+
+{% block content %}
+
+<a href="/content/list/{{primary_category.uuid}}/">
```
Comment (miltontony) [inline cms/templates/content.jinja2]: same here for url. use `request.route_url`

File: cms/templates/molecules.html [hunk 4]
```diff
@@ -0,0 +1,105 @@
+{% import 'atoms.html' as atoms %}
+
+{% macro nav_inline_centered(languages) -%}
```
Comment (miltontony) [inline cms/templates/molecules.html]: can you look at more meaningful names? I'm digging `{% macro banner_with_image(src)` but this isn't making sense.

File: cms/templates/molecules.html [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+</div>
+{%- endmacro %}
+
+{% macro image_and_heading(view, cat) -%}
+{{atoms.image( view.get_image_url(cat.image_host, cat.image, 45, 45), "" , "" ) }}
+<h3>{{cat.title}}</h3>
+{%- endmacro %}
+
+
+{% macro nav_inline_highlight_primary() -%}
```
Comment (miltontony) [inline cms/templates/molecules.html]: Same issue with naming here, I think this should be a more descriptive name. Also, I don't think these blocks need to be hardcoded. Perhaps they should take a list of tuples `(route_name, title)` e.g `site_menu_links([('home', 'Home'), ('about', 'About')])` etc.

File: cms/templates/molecules.html [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+{%- endmacro %}
+
+{% macro footer(message) -%}
+<div class="footer" role="contentinfo">
+  &copy; {{message}}
+</div>
+{%- endmacro %}
+
+{% macro search_box() -%}
+  <form action='/spice/search/' method='GET'>
```
Comment (miltontony) [inline cms/templates/molecules.html]: use `route_path`

File: cms/templates/organisms.html [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+{% endfor %}
+	<li>
+		<a href="/spice/about/{{category.slug}}/">Why it is important</a>
+	</li>
+</ul>
+{%- endmacro %}
+
+{% macro list_of_languages(languages) -%}
+<ul>
+{% for lang in languages[:10] %}
```
Comment (miltontony) [inline cms/templates/organisms.html]: use tuple expansion in loop
```
{% for lang_code, lang_name in languages[:10] %}
<a href="/locale/{{lang_code}}/">{{lang_name}}</a>
{% endfor %}
```

**Comment by nathanbegbie** [comment]
@miltontony Just abstracted the search results and paginator stuff. Have renamed stuff to make the elements clearer (hopefully). I didn't create another template because it fits our model better to use organisms and molecules. Please review

**Comment by miltontony** [comment]
@nathanbegbie I'll look at this over the weekend/monday

**Comment by nathanbegbie** [comment]
@miltontony Finished the template revision stuff and I want to add the sass on now. Should I do that as a separate branch? Either way, pls review :)

File: cms/templates/organisms.html [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    <li>
+    	<a href={{view.request.route_url('category_jinja2', category=cat.uuid)}}>
+    		{{molecules.image_and_heading(view, cat)}}
+       	</a>
+    </li>
+{% endfor %}
+</ul>
+{%- endmacro %}
+
+{% macro list_of_links(view, pages) -%}
```
Comment (miltontony) [inline cms/templates/organisms.html]: This is a bit misleading. `list of links` sounds like it should take a list and render links, instead it's expecting a page and use the `page.title`. I'd recommend either chaning the name or making this more generic by expenging a list of tuples or a dict.

**Comment by miltontony** [comment]
@nathanbegbie looks good. @SteveBarnett when you have a momento, can you have a quick review to ensure we're keeping to the spirit of your research?

**Comment by nathanbegbie** [comment]
@miltontony once Steve has had a look, should I close this branch or keep working on it?

**Comment by miltontony** [comment]
:+1: 
@nathanbegbie New PR for the sass

File: cms/templates/molecules.html [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    <a href={{view.request.route_url('locale_matched',  language=lang_code ) }}>{{lang_name}}</a>
+  </li>
+{% endfor %}
+	<li><a href="/spice/locale/change/">&gt;&gt;</a></li>
+</ul>
+{%- endmacro %}
+
+{% macro banner_with_image(src) -%}
+<div class ="banner">
+	{{ atoms.image(src, "", "")}}
```
Comment (SteveBarnett) [inline cms/templates/molecules.html]: No bigger image here for `srcset`?

File: cms/templates/molecules.html [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+{% macro footer(message) -%}
+<div class="footer" role="contentinfo">
+  &copy; {{message}}
+</div>
+{%- endmacro %}
+
+{% macro search_box(view) -%}
+  <form action={{view.request.route_url('search_jinja')}} method='GET'>
+    <label for="search">Search:
```
Comment (SteveBarnett) [inline cms/templates/molecules.html]: The `for` here and the `id` and `name` of the input should match.

File: cms/templates/molecules.html [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+        <input id="input-search" name="q" type="text">
+    </label>
+        <input type="submit" value='Submit' />
+  </form>
+{%- endmacro %}
+
+{% macro page_number_navigator(view, paginator, p, query) -%}
+
+  {% if paginator.has_previous_page() %}
+    <a href={{view.request.route_url('search_jinja')}}?q={{query}}&p={{p-1}} >&nbsp;&lt;&nbsp;Previous</a>
```
Comment (SteveBarnett) [inline cms/templates/molecules.html]: Are these `&nbsp;`s needed? Use CSS for this kind of spacing.

File: cms/templates/molecules.html [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+  {% if paginator.has_next_page() %}
+    <a href={{view.request.route_url('search_jinja')}}?q={{query}}&p={{p + 1}}>&nbsp;Next&nbsp;&gt;</a>
+  {% endif %}
+
+{%- endmacro %}
+
+{% macro message_for_no_results(query) -%}
+  {% if query %}
+    <p>No results found for <b>{{query}}</b></p>
```
Comment (SteveBarnett) [inline cms/templates/molecules.html]: Do we need to tag these for translation?

**Comment by SteveBarnett** [comment]
@nathanbegbie Woot, looking great! Left a few small comments, otherwise :+1: . :)

File: cms/templates/molecules.html (hunk 9 above)
Comment (nathanbegbie) [inline cms/templates/molecules.html]: There currently isn't one. I assume it will be added later.

File: cms/templates/molecules.html (hunk 12 above)
Comment (nathanbegbie) [inline cms/templates/molecules.html]: Yes, translation will be added later though

File: cms/templates/molecules.html (hunk 9 above)
Comment (miltontony) [inline cms/templates/molecules.html]: @nathanbegbie we can support bigger images. We're currently using `300xh` for the homepage. We can specifiy bigger images using `view.get_image_url(localisation.image_host, localisation.image, 640, 0)`. Please allow for `srcset` with the default being the same as the `src`

**Pull request merged**